source/detail/serialization/xlsx_consumer.cpp

```diff
@@ -1396,7 +1432,15 @@ bool xlsx_consumer::has_cell()
         }
 
         expect_start_element(qn("spreadsheetml", "row"), xml::content::complex); // CT_Row
-        auto row_index = static_cast<row_t>(std::stoul(parser().attribute("r")));
+        row_t row_index = 0;
+        bool ok = detail::parse(parser().attribute("r"), row_index);
+
+        if (!ok)
+        {
+#ifdef THROW_ON_INVALID_XML
```

Comment:
This was actually my initial version, which however produces a compiler warning, and since we turned all warnings into errors, the code will refuse to compile when `THROW_ON_INVALID_XML` is not defined. Which is why I changed the `ifdef` to be placed inside the `if`, hoping that the compiler will identify that the `if` is empty and can be removed.

Reply:
Ah ok, because `ok` is then never used.